### ISSUE TITLE
Hi, i am unable to use the find documentation. Please help.

### ISSUE THREAD

**Issue by shwe2608** [Automattic/mongoose#10610]
I tried to create the sample kittens db. I am unable to use the find function as documented. 

```javascript
const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost:27017/fruitsDB', {useNewUrlParser: true, useUnifiedTopology: true});

const kittySchema = new mongoose.Schema({
  name: String
});

const Kitten = mongoose.model('Kitten', kittySchema);

const silence = new Kitten({ name: 'Silence' });
const fluffy = new Kitten({ name: 'fluffy' });

// silence.save()
// fluffy.save()

Kitten.find(function (err, kittens) {
  if (err) return console.error(err);
  console.log(kittens);
})
```

Getting the following error message in console : 

 node kittens.js
TypeError: cursor.toArray is not a function
    at model.Query.<anonymous> (/Users/shwetank/Desktop/Web Development/fruitsDBProject/node_modules/mongoose/lib/query.js:2151:19)
    at model.Query._wrappedThunk [as _find] (/Users/shwetank/Desktop/Web Development/fruitsDBProject/node_modules/mongoose/lib/helpers/query/wrapThunk.js:27:8)
    at /Users/shwetank/Desktop/Web Development/fruitsDBProject/node_modules/kareem/index.js:370:33
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
(node:30949) UnhandledPromiseRejectionWarning: MongoInvalidArgumentError: Method "collection.find()" accepts at most two arguments
    at Collection.find (/Users/shwetank/Desktop/Web Development/fruitsDBProject/node_modules/mongodb/lib/collection.js:238:19)
    at NativeCollection.<computed> [as find] (/Users/shwetank/Desktop/Web Development/fruitsDBProject/node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js:191:33)
    at NativeCollection.Collection.doQueue (/Users/shwetank/Desktop/Web Development/fruitsDBProject/node_modules/mongoose/lib/collection.js:135:23)
    at /Users/shwetank/Desktop/Web Development/fruitsDBProject/node_modules/mongoose/lib/collection.js:82:24
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:30949) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:30949) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

Please look into it.

**Comment by lorand-horvath** [comment]
> Below script repros the issue. We'll work on a fix tomorrow +1
> 
I already reported it above in https://github.com/Automattic/mongoose/issues/10610#issuecomment-907060604
including the repro script, see it again below - I guess it wasn't clear enough? Only difference was that I used an IIFE instead:
```javascript
mongoose.connect(DB);
(async () => {
    const users = await User.find();
    console.log(users.length);
})().catch(err => console.log(err));
```

**Comment by Mgotya** [comment]
When this fix will be released? it fixed two day ago but the 6.0.2 release was three days ago so this fix is not a part of 6.0.2

**Comment by vkarpov15** [comment]
@Mgotya we will release 6.0.3 with 9de60a7 later today.
